driver.go

```diff
@@ -47,6 +48,12 @@ type QueryMutator interface {
 	MutateQuery(ctx context.Context, req backend.DataQuery) (context.Context, backend.DataQuery)
 }
 
+// HeaderMutator is an additional interface that could be implemented by driver.
+// This adds the ability to mutate headers before they are forwarded to the driver.
+type HeaderMutator interface {
+	MutateHeaders(ctx context.Context, headers http.Header) http.Header
```

Comment:
I didn't think it necessary to return `ctx` from `MutateHeaders` like the other mutator interfaces because I don't understand under what scenario a datasource would want to change the context passed to it.